### PR TITLE
Chef Server required fixups

### PR DESCRIPTION
**Pull request by sersut**
This PR fixes two issues with license_scout encountered while collecting licenses for Chef Server and adds required overrides for Chef Server.

/cc: @danielsdeleo @ryancragun

### CODE REVIEW

**Comment by sersut** [comment]
@danielsdeleo I also pushed https://github.com/chef/license_scout/commit/9fa7b97048cd0b7afc9aada02be01edc96a394b5 to this branch. Without this opscode-expander under chef-server fails not finding some gems because they are installed under a different path, because of this:

https://github.com/chef/chef-server/blob/master/omnibus/config/software/opscode-expander.rb#L28

After much research this seems to be the best way and I regression tested this on Chef Client:

http://manhattan.ci.chef.co/job/chef-build/344/

File: spec/license_scout/dependency_manager/bundler_spec.rb [hunk 1]
```diff
@@ -76,13 +76,10 @@
       FileUtils.touch(File.join(project_dir, "Gemfile"))
     end
 
-    it "detects the project as a bundler project" do
-      expect(bundler.detected?).to eq(true)
+    it "does not detect the project as a bundler project" do
```
Comment (danielsdeleo) [inline spec/license_scout/dependency_manager/bundler_spec.rb]: Do we need one of these for when there's a gemfile and lockfile?

Comment (sersut) [inline spec/license_scout/dependency_manager/bundler_spec.rb]: Ah yes. Adding this now.

**Comment by sersut** [comment]
@danielsdeleo I'm merging this based on our chat today in order to have a clean PR for berkshelf backend. LMK if there is anything else needed.

**Pull request merged**